remappers/internal/metric.go

```diff
@@ -76,3 +78,55 @@ func AddMetrics(
 		mutator(dp)
 	}
 }
+
+func Addk8sMetrics(ms pmetric.MetricSlice, mutator func(dp pmetric.NumberDataPoint), metrics ...Metric) {
```

Comment:
I think we can use the `AddMetrics` here. The kubernetes specific logic should be added in the `mutator` and be kept isolated in the `kubernetesmetrics` pacakge.

Reply:
I see that the kubernetes specific logic here is setting dataset which is dependent on the metric name. This would not be possible to set using the current design of the mutator which only takes a datapoint. There are 2 options we could consider:

1. Update the `Metric` model to have a `Dataset` field and change the signature of the `AddMetric` to `func(ms pmetric.MetricSlice, mutator func(dp pmetric.NumberDataPoint), metrics ...Metric)`. Then each metric can have its own dataset. Note that we would need to consider the config option and set empty dataset if the flag `WithKubernetesIntegrationDataset` is not set.
2. Update the design of the mutator to take to `func(m pmetric.Metric)` and use the mutator to do the k8s specific dataset logic.

I would personally prefer `1` because it is cleaner IMO and dataset is specific to each metric by our design but I will leave it up to you to decide which is better.